pkg/apis/pipeline/reservedsidecar.go

```diff
@@ -0,0 +1,18 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pipeline
+
+// ReservedResultsSidecarName is the name of the results sidecar that outputs the results to stdout
+// when the results-from feature-flag is set to "sidecar-logs".
+const ReservedResultsSidecarName = "tekton-log-results"
```

Comment:
Perhaps this should live in `pkg/pod` instead of `pkg/pipeline`? Most of the consts in `pkg/pipeline` deal with values that users may need to interact with in Tasks, while this feels like an internal implementation detail of how we construct pods.

Reply:
good call! Done.

Reply:
Actually, I take it back. The reason I added it `pkg/pipeline` is because I need for validation for admission web hook where I ensure that the users don't provide the same name as that reserved by this. By keeping it in pkg/pod I was running into circular dependency issues. Because pkg/pod imports pkg/pipeline/v1beta1 and vice versa.

Reply:
The part about admission webhook is in the very last PR: https://github.com/tektoncd/pipeline/pull/5695

Reply:
Let me know if `pkg/pipeline` is better in this case and I will move it back.